src/moscot/backends/ott/solver.py

```diff
@@ -97,6 +98,11 @@ def _create_geometry(
             if is_linear_term:
                 self._graph_in_linear_term = True
             if x.cost == "geodesic":
+                if n_src_tgt is not None:
+                    n_src, n_tgt = n_src_tgt
+                    if n_src + n_tgt != arr.shape[0]:
+                        raise ValueError(f"Expected `x` to have `{n_src + n_tgt}` points, found `{arr.shape[0]}`.")
+                    arr = arr[:n_src, n_src:]
```

Comment:
@MUCDK do you think this makes sense? I think yes cause the indices are checked here https://github.com/theislab/moscot/blob/90600424297d6ec612756d9d2f52c95463c90ea4/src/moscot/base/problems/problem.py#L708

Reply:
I don't think it makes sense because of another reason: https://github.com/ott-jax/ott/issues/479#issuecomment-1860887301